src/user-event/utils/wait.ts

```diff
@@ -0,0 +1,15 @@
+import { UserEventConfig } from '../setup';
+
+export function wait(config: UserEventConfig) {
+  const delay = config.delay;
+  if (typeof delay !== 'number') {
+    return;
+  }
+
+  return Promise.all([
```

Comment:
This is very confusing to me, I don't understand really why the advanceTimers function is used only for fake timers. Here we wait for two promises, one being useful in case fake timers are used and the other in case real timers are used, is this because we don't want to use `jestFakeTimersAreEnabled `?

Also I don't really see the point of having  `advanceTimers` as an option, what could it be used for? We don't really support any other test runner aside from jest currently right? I don't really know what users are using as a test runner but I have the impression that in RN with all the required presets Jest is currently the only solution that's supported by most major libraries

Reply:
This implementation is taken straight from actual User Event code: https://github.com/testing-library/user-event/blob/main/src/utils/misc/wait.ts

The way it works is that the first part: `new Promise<void>(resolve => globalThis.setTimeout(() => resolve(), delay)),` works for both real and fake timers.

The second part `config.advanceTimers(delay)` is configurable through `userEvent.setup()` call. By default UserEvent has `() => Promise.resolve()` there, which allows this function to work with real timers. Having it as an option allows to support different fake timer types, e.g. [Sinon.JS](https://sinonjs.org/releases/latest/fake-timers/) and not just Jest ones.

My innovation in that regard is that I've created a [universal `advanceTimers` default](https://github.com/callstack/react-native-testing-library/pull/1405/files#diff-64635189362ce288644481b4ecbcfd937bd623c96831718a6d77e04ef460e591) function that works with both real timers and fake Jest timers, which I assume are used by majority of our users using fake timers at all.

Having that as an option allows other fake timers users, still to use them with our User Event. And there have been [tickets](https://github.com/callstack/react-native-testing-library/pull/1169) from users using Sinon.JS as more powerful timers implementation.

Reply:
Ok this seems reasonable if we want to support other fake timers than the ones from Jest. I'm curious though is there a way in user-event to configure the advanceTimers option globally or does it need to be done every time in the setup function?

Reply:
For global, RTL recommends putting User Event setup into custom render function:

```
// setup function
function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  }
}

test('render with a setup function', async () => {
  const {user} = setup(<MyComponent />)
  // ...
})
```

Source: https://testing-library.com/docs/user-event/intro#writing-tests-with-userevent